### PR TITLE
adding max number of attempts

### LINKED ISSUE #7: Add retry limit
Currently `solve` will attempt to request for a solution for as long as the response is `{"status":0,"request":"CAPCHA_NOT_READY"}`. I have observed cases where this has persisted for a very long time. It would be handy to have a way to configure a hard limit or to have `cancel` method.

### PR DESCRIPTION
**Pull request by pedroapfilho**
Fixes #7

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -59,18 +69,20 @@ const captchaSolver = (key: string) => {
...
+      attempts -= 1;
+
+      return getCaptcha(id, attempts);
     } catch (e) {
       throw new Error(e);
     }
   };
 
-  const solveCaptcha = async (image: string) => {
+  const solveCaptcha = async (image: string, maxAttempts: number = 20) => {
```
Comment (gajus) [inline src/index.ts]: I would use an object for configuration. There are bunch of other properties that 2captcha supports at a time of requesting a solution. See https://2captcha.com/2captcha-api#solving_captchas.

File: src/index.ts [hunk 2]
```diff
@@ -59,18 +69,20 @@ const captchaSolver = (key: string) => {
 
       await timer(1000);
 
-      return getCaptcha(id);
+      attempts -= 1;
+
+      return getCaptcha(id, attempts);
```
Comment (gajus) [inline src/index.ts]: `attempts` seems redundant. Just pass `maxAttempts - 1`.

File: src/index.ts (hunk 1 above)
Comment (pedroapfilho) [reply src/index.ts]: Fixed, I created an Interface for this one too, it's gonna be easier to maintain.

File: src/index.ts (hunk 2 above)
Comment (pedroapfilho) [reply src/index.ts]: Ya, you're right.. fixed.

**Pull request merged**